src/main/java/com/amazon/opendistroforelasticsearch/security/ssl/transport/OpenDistroSecuritySSLRequestHandler.java

```diff
@@ -100,6 +101,13 @@ public final void messageReceived(T request, TransportChannel channel, Task task
             channel.sendResponse(exception);
             throw exception;
         }
+
+        TransportChannel innerChannel = channel;
+        if("PerformanceAnalyzerTransportChannelType".equals(channel.getChannelType())) {
+            Class patc = channel.getClass();
+            Method getInnerChannel = patc.getMethod("getInnerChannel", null);
+            innerChannel = (TransportChannel) getInnerChannel.invoke(channel);
```

Comment:
getInnerChannel can be null if "getInnerChannel" method is not implemented by Performance Analyzer plugin. And it can through null pointer exception.
Can we handle null here?

Reply:
Makes sense. Will update the PR